doc/cli-directive.md

```diff
@@ -203,6 +203,12 @@ cli:
         split-operation-names:
           - Create
           - Update
+    # flatten parameter
+      - where:
+          group: OperationGroupName
+          op: CreateOrUpdate#Update
+        flatten-params:
+          - properties
```

Comment:
please follow the current pattern like below. And it's good to use cli-flatten instead of flatten-param so that you dont need to introduce another flatten-xxx later in case we need to support flatten other things. Feel free to raise other general name you think better than cli-flatten
- where:
      group: OperationGroupName
      op: ...
      param: properties
  cli-flatten: true

Reply:
I agree to reuse current flatten concept, but there is one concern for adding `cli-flatten` on property directly.  After modelerfour, request's parameter becomes virtual parameter. For different requests with same schema, they share the instance. It means if directive add cli-flatten on one parameter, It will make me flatten un-expected parameter. For other plugins they have no such issue, because flattenSetter add directive before modelerfour and poly clone the request.

Reply:
So I think we cannot add directive on target parameter. How about remove 
``` yaml
    flatten-param:
        cli-flatten-param-enabled: true
```
Then use `cli-flatten-set-enabled: true` as switch. But we still need `flatten-params` on operation

Reply:
Do you mean the parameter is shared after m4? or property? it's expected for property to be shared, we are adding directives on parameters, it will be a big problem if it's shared. I remember VirtualParameter is created when m4 doing flatten, i dont think it's shared, could you confirm? thx

Reply:
Confirmed again. They do shared. It happens only in request parameter.

Reply:
Fixed. Thanks